src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double DEFAULT_RADIUS = 10.0;
+    private static final int NUMBER_OF_FIGURES = 5;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public BaseShape getRandomFigure() {
+        int figureType = random.nextInt(NUMBER_OF_FIGURES);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                double squareSide = random.nextDouble() * 10;
```

Comment:
create separate private method for `random.nextDouble() * 10` and use it for each type of figure